docker/importer/importer.py

```diff
@@ -452,8 +456,65 @@ def thread_init():
 
     logging.info("Finished processing bucket: %s", source_repo.name)
 
+  def _process_updates_rest(self, source_repo: osv.SourceRepository):
+    """Process updates from REST API."""
+    logging.info('Begin processing REST: %s', source_repo.name)
+    import_time_now = utcnow()
+    request = requests.head(source_repo.rest_api_url, timeout=_TIMEOUT_SECONDS)
+    if request.status_code != 200:
+      logging.error('Failed to fetch REST API: %s', request.status_code)
+      return
+    last_modified = datetime.datetime.strptime(request.headers['Last-Modified'],
+                                               _HTTP_LAST_MODIFIED_FORMAT)
+    # Check whether endpoint has been modified since last update
+    if last_modified < source_repo.last_update_date:
+      logging.info('No changes since last update.')
+      return
+    request = requests.get(source_repo.rest_api_url, timeout=_TIMEOUT_SECONDS)
+    # Get all vulnerabilities from the REST API. (CURL approach)
+    vulns = request.json()
+    # Create tasks for changed files.
+    for vuln in vulns:
+      import_failure_logs = []
+      last_modified = datetime.datetime.strptime(vuln['modified'],
+                                                 _OSV_MODIFIED_FORMAT)
+
+      if last_modified < source_repo.last_update_date:
+        continue
+      try:
+        #TODO(jesslowe): Use a ThreadPoolExecutor to parallelize this
+        vuln_location = source_repo.link + vuln['id'] + source_repo.extension
+        single_request = requests.get(vuln_location, timeout=_TIMEOUT_SECONDS)
```

Comment:
nit: `single_vuln` would be more descriptive as a variable name. `single_request` can be confusing as the variable itself is not exactly a "request", but rather the response (i.e. the vuln).

Reply:
Yeah, that works a lot better. Changed :)